cpp/src/arrow/array.h

```diff
@@ -39,7 +40,7 @@ class Array {
   Array(const std::shared_ptr<DataType>& type, int32_t length, int32_t null_count = 0,
       const std::shared_ptr<Buffer>& null_bitmap = nullptr);
 
-  virtual ~Array() {}
+  virtual ~Array() = default;
```

Comment:
I changed to default purely for style (I likely should have done this as a separate PR and discussed), for some reason it reads more elegantly to me.